src/Validator.php

```diff
@@ -34,10 +35,21 @@ public function __construct(private RuleHandlerResolverInterface $ruleHandlerRes
     public function validate(mixed $data, ?iterable $rules = null): Result
     {
         $data = $this->normalizeDataSet($data);
+
+        $rulesByAttribute = [];
+
+        if ($data instanceof DataSetInterface) {
```

Comment:
Attributes are supported in non-DataSetInterface.

Reply:
The check should be is_object.

Reply:
done.